lib/CTownHandler.h

```diff
@@ -122,7 +122,7 @@ class DLL_LINKAGE CBuilding
 			h & subId;
 			h & height;
 		}
-		else if(!h.saving)
+		if(!h.saving && version < 793)

```

Comment:
Do not change version in this case, it is just a fix.
```suggestion
		if(!h.saving && version < 792)

```

Reply:
Maybe <= 792, because of new added visiting bonuses - we should set subID to make them working
But in this case there will be no difference between saves with visiting SubID-s  and without them

Reply:
I still do not understand.
1. Why version has been increased in this PR?
2. Why are you updating subId for version 792?

Reply:
I added new values for subId in Part II : some visiting bonuses and others

Suppose, user have save after Part I , but before Part II,
so there're no data about buildings, added in the Part II, but the serializer version is same as in the Part I - 792
Then he update VCMI, update mod, adding new supported types,  load save and note, that visiting bonus is not working.

It's because there's "NONE" subId instead of subId-s added in the Part II and update procedure did not work
because serializer version is actual

Thus, I suggest call `update792` and `updateBonusingBuildings` again to fill visiting bonuses with corresponding Ids 
and increase the serializer version to avoid unnecessary calls (when all Ids should  already present in the save)